### PR TITLE
[SYCL] Change SYCL version according to standard

### PR DESCRIPTION
**Pull request by dklochkov-emb**
Version of SYCL was changed according to the latest agreement:
The lower 2 digits are not formally specified, but we plan to use these to identify the month in which we submit the specification for ratification, which is similar to the C++ macro __cplusplus.

Since the SYCL 2020 specification was submitted for ratification in December of 2020, the macro's value is now 202012 for SYCL 2020.
see PR for details 
https://github.com/KhronosGroup/SYCL-Docs/pull/634

### CODE REVIEW

File: clang/lib/Frontend/InitPreprocessor.cpp [hunk 1]
```diff
@@ -587,7 +587,7 @@ static void InitializeStandardPredefinedMacros(const TargetInfo &TI,
     if (LangOpts.getSYCLVersion() == LangOptions::SYCL_2017)
       Builder.defineMacro("CL_SYCL_LANGUAGE_VERSION", "121");
     else if (LangOpts.getSYCLVersion() == LangOptions::SYCL_2020)
-      Builder.defineMacro("SYCL_LANGUAGE_VERSION", "202001");
+      Builder.defineMacro("SYCL_LANGUAGE_VERSION", "202012");
```
Comment (gmlueck) [inline clang/lib/Frontend/InitPreprocessor.cpp:590]: I don't know this code well, but it seems strange that the `L` suffix is missing.  Should it be like this instead?

```suggestion
      Builder.defineMacro("SYCL_LANGUAGE_VERSION", "202012L");
```

Comment (gmlueck) [reply clang/lib/Frontend/InitPreprocessor.cpp]: Thanks.  Seems like this is resolved now.

**Comment by keryell** [comment]
That sounds good.
Could you fix the typos in the PR title and the commit messages.
Please write SYCL in uppercase everywhere since it is a standard name.

**Comment by keryell** [comment]
> That sounds good. Could you fix the typos in the PR title and the commit messages. Please write SYCL in uppercase everywhere since it is a standard name.

For the typos I mean `accrodnig` → `according` in both the commit message and the PR title. :smile:

**Review by bader — APPROVED** [review]

**Review by gmlueck — APPROVED** [review]

**Pull request merged**